### PR TITLE
Update go.mods to use latest version of k8s.io/client-go

### PR DESCRIPTION
**Pull request by tingrui-AWS-EKS-auth**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: tests/e2e/go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/onsi/gomega v1.23.0
 	k8s.io/api v0.26.1
 	k8s.io/apimachinery v0.26.1
-	k8s.io/client-go v0.26.1
+	k8s.io/client-go v1.5.2
```
Comment (dims) [inline tests/e2e/go.mod:10]: Can you use `v0.27.4` here as well?

Comment (tingrui-AWS-EKS-auth) [reply tests/e2e/go.mod]: using v0.27.4 now.

**Comment by nckturner** [comment]
@tingrui-AWS-EKS-auth I think you are missing a corresponding change in a go.sum file:  

```
go: k8s.io/api@v0.27.4: missing go.sum entry for go.mod file; to add it:
	go mod download k8s.io/api
go: downloading k8s.io/kubernetes v1.26.1
go: k8s.io/api@v0.27.4: missing go.sum entry for go.mod file; to add it:
	go mod download k8s.io/api
Cleaning up test role aws-iam-authenticator-test-role-6394
make: *** [Makefile:119: integration] Error 1
+ EXIT_VALUE=2
+ set +o xtrace
```

**Pull request closed without merging**